hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/TypedTable.java

```diff
@@ -87,32 +88,43 @@ public TypedTable(RDBTable rawTable,
    * @param keyType The key type.
    * @param valueType The value type.
    * @param cacheType How to cache the entries?
-   * @param threadNamePrefix
+   * @param threadNamePrefix the thread name prefix
    * @throws IOException if failed to iterate the raw table.
    */
   public TypedTable(RDBTable rawTable,
       CodecRegistry codecRegistry, Class<KEY> keyType,
       Class<VALUE> valueType,
       CacheType cacheType, String threadNamePrefix) throws IOException {
-    this.rawTable = Objects.requireNonNull(rawTable, "rawTable==null");
-    Objects.requireNonNull(codecRegistry, "codecRegistry == null");
+    this(rawTable, keyType, codecRegistry.getCodecFromClass(keyType), valueType,
+        codecRegistry.getCodecFromClass(valueType), cacheType, threadNamePrefix);
+  }
 
+  /**
+   * Create an TypedTable from the raw table with specified cache type.
+   *
+   * @param rawTable The underlying (untyped) table in RocksDB.
+   * @param keyType The key type.
+   * @param keyCodec The key codec.
+   * @param valueType The value type.
+   * @param valueCodec The value codec.
+   * @param cacheType How to cache the entries?
+   * @param threadNamePrefix the thread name prefix
+   * @throws IOException
+   */
+  public TypedTable(
+      RDBTable rawTable, Class<KEY> keyType, Codec<KEY> keyCodec, Class<VALUE> valueType, Codec<VALUE> valueCodec,
```

Comment:
Remove `keyType`  and use `keyCodec.getTypeClass()`.

Do the same for `valueType`.